eclair-front/src/test/scala/fr/acinq/eclair/router/FrontRouterSpec.scala

```diff
@@ -149,9 +148,18 @@ class FrontRouterSpec extends TestKit(ActorSystem("test")) with AnyFunSuiteLike
     val system2 = ActorSystem("front-system-2")
     val system3 = ActorSystem("front-system-3")
 
-    val front1 = system1.actorOf(FrontRouter.props(nodeParams.routerConf, router))
-    val front2 = system2.actorOf(FrontRouter.props(nodeParams.routerConf, router))
-    val front3 = system3.actorOf(FrontRouter.props(nodeParams.routerConf, router))
+    val front1 = {
+      implicit val system: ActorSystem = system1
+      TestFSMRef[FrontRouter.State, FrontRouter.Data, FrontRouter](new FrontRouter(nodeParams.routerConf, router))
+    }
+    val front2 = {
+      implicit val system: ActorSystem = system2
+      TestFSMRef[FrontRouter.State, FrontRouter.Data, FrontRouter](new FrontRouter(nodeParams.routerConf, router))
+    }
+    val front3 = {
+      implicit val system: ActorSystem = system3
+      TestFSMRef[FrontRouter.State, FrontRouter.Data, FrontRouter](new FrontRouter(nodeParams.routerConf, router))
+    }
```

Comment:
I just realized that this trick (providing a custom `ActorSystem`) would also make it possible to segregate event streams in tests. Should I do that instead of [40cc09f](https://github.com/ACINQ/eclair/pull/2268/commits/40cc09f102a0d0bc4b485061c4ea5e000174e6cf)? I think I should 😅